_posts/2018-08-06-bashing-through-minikube.markdown

```diff
@@ -0,0 +1,91 @@
+
+### 3. Have Your Container Entrypoint And Command Be Targeted
+The containers that you are expecting to run in your cluster should be single purpose by design, that means that you can have your entrypoint and command set so you don't have to manage that outside the Dockerfile itself.
+
+### 4. Exposing Ports
+#### e.g. I want to see my postgres database
+`kubectl port-forward $(kubectl get pods | grep db | awk {'print $1'}) 5432:5432`
+
+This is an example command for accessing a Postgres database on the cluster. If we break it down:
+1. `kubectl port-forward {pod-name} source-port:destination-port`
+
+    This command sets up the port forwarding so the given pod port is accessible on `localhost:destination-port`
+1. `kubectl get pods | grep db | aws {'print $1'}`
+
+    This command lists all the pods, selects the line that has the term `db` and then prints the first column of data, i.e. the pod name.
+
+### 5. Alias Away Common Command Chains
+Minikube, alas, is not perfect which means you have to restart it from scratch on occasion. Also when iterating on the scripts from pattern #1 it's helpful to make sure they work on a fresh machine.
+
+`alias reset-minikube='minikube delete && minikube start && helm init`
+
+Also for iterating on your integration test feedback loop:
+
+`alias warm-reload='./local-nuke.sh && ./local-build.sh && ./local-run.sh'`
+
+And for managing the docker environments:
+
+`alias local-docker='unset DOCKER_API_VERSION DOCKER_TLS_VERIFY DOCKER_CERT_PATH DOCKER_API_VERSION DOCKER_HOST'`
+
+## Anti-Patterns
+### 1. Every Project In A Repository Should Be Testable In Isolation
+With Docker it is so trivial to spin up a database container that people can make the mistake of making all their component tests require an actual database to function. This complicates matters significantly especially when you add parallel docker environments between minikube and your local machine.
+
+### 2. Don't Rely On Being Able To Run Your Entire Stack Locally
+When you have a small number of services it is possible to run them all locally at once, however this stops being the case quickly and is more difficult to retroactively fix then to stop this at the start.
+
+The need to do this is generally driven by insufficient testing, or testing at the wrong isolation levels. For testing across repositories leave that to Continuous Delivery.
+
+## Wrap Up
+Hopefully this post will have saved you many frustrations in the road ahead to Kubernetes adoption. Minikube is fantastic for keeping your feedback cycles fast and learning its quirks is well worth the time investment.
+
+The next step is getting to grips with your Kubernetes flavour of choice, be that cloud vendor managed or for those stout of heart a bare metal installation. Stay tuned for posts on adopting EKS, up-skilling the team into Kubernetes, and making your cluster production ready.
```

Comment:
can you provide a link to EKS?

Stay tuned for posts on adopting [EKS](https://aws.amazon.com/eks/)